### PR TITLE
Add MOI methods: delete and modify

### PR DESCRIPTION
**Pull request by laradicp**

### CODE REVIEW

File: src/MOIwrapper.jl [hunk 1]
```diff
@@ -207,6 +207,48 @@ function MOI.add_constraint(
...
+)
+    constrid = getid(model.constrs[ci])
+    orig_form = model.inner.original_formulation
+    coefmatrix = getcoefmatrix(orig_form)
+    varids = VarId[]
+    for (varid, _) in @view coefmatrix[constrid, :]
+        push!(varids, varid)
+    end
+    for varid in varids
+        coefmatrix[constrid, varid] = 0.0
```
Comment (guimarqu) [inline src/MOIwrapper.jl:224]: What happens if you do :

```julia
for (varid, _) in @view coefmatrix[constrid, :]
        coefmatrix[constrid, varid] = 0.0
end
```

Comment (guimarqu) [reply src/MOIwrapper.jl:224]: I think it enters in a undefined behavior because the size of the array that stores the coefficient matrix may change.

**Review by guimarqu — COMMENTED** [review]
I suggest to create methods to delete variable and constraint in MathProg.

So we would have :

```
    delete!(formulation, var)
    delete!(formulation, varid)
    delete!(formulation, constr)
    delete!(formulation, constrid)
```

File: src/MOIwrapper.jl [hunk 2]
```diff
@@ -209,6 +209,89 @@ function MOI.add_constraint(
...
+    return
+end
+
+function MOI.modify(
+    model::Coluna.Optimizer, ::MOI.ObjectiveFunction{MOI.ScalarAffineFunction{Float64}},
+    change::MathOptInterface.ScalarCoefficientChange{Float64}
+)
+    var = model.vars[change.variable]
+    var.perendata.cost = change.new_coefficient
+    var.curdata.cost = change.new_coefficient
```
Comment (guimarqu) [inline src/MOIwrapper.jl:242]: You should use `setperencost` here otherwise changed are not propagated to the subsolver.

File: src/MOIwrapper.jl [hunk 3]
```diff
@@ -209,6 +209,89 @@ function MOI.add_constraint(
...
+    return
+end
+
+function MOI.modify(
+    model::Coluna.Optimizer, ci::MOI.ConstraintIndex{F,S},
+    change::MOI.ScalarConstantChange{Float64}
+) where {F<:MOI.ScalarAffineFunction{Float64},S}
+    constr = model.constrs[ci]
+    constr.perendata.rhs = change.new_constant
+    constr.curdata.rhs = change.new_constant
```
Comment (guimarqu) [inline src/MOIwrapper.jl:281]: You shoud use `setperenrhs!`.

File: src/MOIwrapper.jl [hunk 4]
```diff
@@ -209,6 +209,89 @@ function MOI.add_constraint(
...
+    return
+end
+
+function MOI.modify(
+    model::Coluna.Optimizer, ci::MOI.ConstraintIndex{F,S},
+    change::MOI.ScalarCoefficientChange{Float64}
+) where {F<:MOI.ScalarAffineFunction{Float64},S}
+    varid = getid(model.vars[change.variable])
+    constrid = getid(model.constrs[ci])
+    getcoefmatrix(model.inner.original_formulation)[constrid, varid] = change.new_coefficient
```
Comment (guimarqu) [inline src/MOIwrapper.jl:291]: it's better to retrieve the original formulation with `get_original_formulation(model)`.

File: src/MathProg/varconstr.jl [hunk 5]
```diff
@@ -453,6 +453,26 @@ function deactivate!(form::Formulation, f::Function)
...
+function Base.delete!(form::Formulation, varid::VarId)
+    delete!(form.manager.vars, varid)
+    delete!(form.buffer.var_buffer.added, varid)
+    return
+end
+Base.delete!(form::Formulation, var::Variable) = delete!(form, getid(var))
+
+function Base.delete!(form::Formulation, constrid::ConstrId)
+    delete!(form.buffer.constr_buffer.added, constrid)
+    delete!(form.manager.constrs, constrid)
```
Comment (guimarqu) [inline src/MathProg/varconstr.jl:472]:
```suggestion
    delete!(form.manager.constrs, constrid)
    return
```

File: src/MOIwrapper.jl [hunk 6]
```diff
@@ -209,6 +209,84 @@ function MOI.add_constraint(
...
+function MOI.delete(
+    model::Coluna.Optimizer, ci::MOI.ConstraintIndex{F,S}
+) where {F<:MOI.ScalarAffineFunction{Float64},S}
+    constrid = getid(model.constrs[ci])
+    orig_form = get_original_formulation(model.inner)
+    coefmatrix = getcoefmatrix(orig_form)
+    varids = VarId[]
+    for (varid, _) in @view coefmatrix[constrid, :]
+        push!(varids, varid)
+    end
```
Comment (guimarqu) [inline src/MOIwrapper.jl:263]: I think you should change the coefficients in `MathProg.delete!`.

**Comment by guimarqu** [comment]
I need these changes to fix #583 so I merge into master.

File: src/MOIwrapper.jl (hunk 6 above)
Comment (laradicp) [reply src/MOIwrapper.jl:263]: Ah indeed

**Pull request merged**